sw/device/lib/dif/dif_template.h.tpl

```diff
@@ -46,273 +43,275 @@ extern "C" {
 
 /**
- * Hardware instantiation parameters for <peripheral>.
+ * Hardware instantiation parameters for ${periph_lower}.
  *
  * This struct describes information about the underlying hardware that is
  * not determined until the hardware design is used as part of a top-level
  * design.
  */
-typedef struct dif_<ip>_param {
+typedef struct dif_${ip_snake}_params {
   /**
-   * The base address for the <peripheral> hardware registers.
+   * The base address for the ${periph_lower} hardware registers.
    */
   mmio_region_t base_addr;
 
   // Other fields, if necessary.
-} dif_<ip>_param_t;
+} dif_${ip_snake}_params_t;
 
 /**
- * Runtime configuration for <peripheral>.
+ * Runtime configuration for ${periph_lower}.
  *
  * This struct describes runtime information for one-time configuration of the
  * hardware.
  */
-typedef struct dif_<ip>_config {
+typedef struct dif_${ip_snake}_config {
   // Fields, if necessary.
-} dif_<ip>_config_t;
+} dif_${ip_snake}_config_t;
 
 /**
- * A handle to <peripheral>.
+ * A handle to ${periph_lower}.
  *
  * This type should be treated as opaque by users.
  */
-typedef struct dif_<ip> {
-  dif_param_<ip>_t params;
+typedef struct dif_${ip_snake} {
+  dif_${ip_snake}_params_t params;
 
   // Other fields, if necessary.
-} dif_<ip>_t;
+} dif_${ip_snake}_t;
 
 /**
- * The result of a <peripheral> operation.
+ * The result of a ${periph_lower} operation.
  */
-typedef enum dif_<ip>_result {
+typedef enum dif_${ip_snake}_result {
   /**
    * Indicates that the operation succeeded.
    */
-  kDif<IP>Ok = 0,
+  kDif${ip_camel}Ok = 0,
   /**
    * Indicates some unspecified failure.
    */
-  kDif<IP>Error = 1,
+  kDif${ip_camel}Error = 1,
   /**
    * Indicates that some parameter passed into a function failed a
    * precondition.
    *
    * When this value is returned, no hardware operations occured.
    */
-  kDif<IP>BadArg = 2,
+  kDif${ip_camel}BadArg = 2,
   /**
    * Indicates that this operation has been locked out, and can never
    * succeed until hardware reset.
    */
   // Remove this variant if you don't need it.
-  kDif<IP>Locked = 3,
-} dif_<ip>_result_t;
+  kDif${ip_camel}Locked = 3,
+} dif_${ip_snake}_result_t;
 
 
 /**
- * Parameters for a <peripheral> transaction.
+ * Parameters for a ${periph_lower} transaction.
  */
-typedef struct dif_<ip>_transaction {
+typedef struct dif_${ip_snake}_transaction {
   // Your fields here.
-} dif_<ip>_transaction_t;
+} dif_${ip_snake}_transaction_t;
 
 /**
- * An output location for a <peripheral> transaction.
+ * An output location for a ${periph_lower} transaction.
  */
-typedef struct dif_<ip>_output {
+typedef struct dif_${ip_snake}_output {
   // Your fields here.
-} dif_<ip>_output_t;
+} dif_${ip_snake}_output_t;
 
 /**
- * A <peripheral> interrupt request type.
+ * A ${periph_lower} interrupt request type.
  */
-typedef enum dif_<ip>_irq {
+typedef enum dif_${ip_snake}_irq {
   // Your IRQs here!
-} dif_<ip>_irq_t;
+} dif_${ip_snake}_irq_t;
```

Comment:
If you parse the IP HJSon as another input to your script, you can populate this automatically as well.

Reply:
I feel like that's a step too far in complexity; we haven't actually decided that we want to use the IP name every time, anyway. This seems like something for a followup.